icu4c/source/i18n/chnsecal.cpp

```diff
@@ -333,6 +338,14 @@ struct MonthInfo computeMonthInfo(
  * @stable ICU 2.8
  */
 int64_t ChineseCalendar::handleComputeMonthStart(int32_t eyear, int32_t month, UBool useMonth, UErrorCode& status) const {
```

Comment:
Same question here: Is it really worth it to split `handleComputeMonthStart()` into two functions like this?

Reply:
It is necessary because we cannot change the signature of handleComputeMonthStart since it is defined by the superclass but we need to pass in the leap information separately to avoid infinity recusion